### PR TITLE
Added Pre-Upgrade report endpoint

### PR DESCRIPTION
**Pull request by artmello**
- Added support for calling pre_upgrade_report actor passing no data

### CODE REVIEW

**Comment by artmello** [comment]
Depending on https://github.com/leapp-to/leapp-actors/pull/19

**Comment by artmello** [comment]
Following the idea proposed on latest PR against leappctl, where client would receive a command like:
$ leappctl check-system --checks pre_upgrade

Intention would be to support, in a "standard" manner, either pre_upgrade checks, pre_migration checks or any other check for which an actor is written.

I could imagine three scenarios:
* First one, leappctl would parse user input and call the specific daemon endpoint for that specific check actor. On that scenario we would need to have one endpoint written and modified for each check actor;
* Another option would be having a single check-system endpoint, calling a check-system actor and making such an actor decide (via input message channel) which actor/group to call. I don't think that would be simple to implement or to add new checks with current actor yaml;
* Final one, the one implemented here, is to have a single check-system endpoint that will parse client input and call the requested actor. I choose this one because we can write new check actors and call them without changing daemon or leappctl.

What do you guys think?
cc/ @bocekm @bertinatto @vinzenz

**Comment by vinzenz** [comment]
To be honest I am confused about "check_html_output" - This will be used to return the formatted HTML to the clients, if I understand that correct, right? And the clients are sending the data to the daemon which then gets formatted and returned again as a JSON response, is that about right?

**Comment by artmello** [comment]
@vinzenz check_html_output is an actor that will apply a jinja template and output HTML from data using CheckOutput schema (or data generated by a "check module" actor). Instead of this being a standalone tool called by leappctl or even code inside leappctl, I implemented it as an actor. I am not adding such actor by default into the "check modules group" actor because it sounds to me that HTML output is not something you will want every time. As an actor, called via leapp-daemon, it was not clear to me if it was "safe" to try to save HTML output into a file, since I am not sure if leapp-daemon/actor would always have access to write into some path that the user have access to read. That is why HTML result is being return as a string output from the actor and leappctl is saving that into a file (change made in a leappctl PR)

File: pkg/api/check_html_output.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+	Summary string   `json:"summary"`
+	Params  []string `json:"params"`
+}
+
+type checks struct {
+	Checks []checkEntry `json:"checks"`
+}
+
+type checkHTMLOutputParams struct {
+	CheckOutput []checks `json:"check_output"`
```
Comment (bertinatto) [inline pkg/api/check_html_output.go:21]: In hindsight, it'd be better to use `values` here [1], so we could use `ObjValue` in this struct. 

I think it's OK to change this in another opportunity, though.

[1] https://github.com/leapp-to/leapp-actors/pull/19/files#diff-6b3a6f81f69e8399e64fe61b9c2e5186R17

**Review by bertinatto — APPROVED** [review]

**Comment by bertinatto** [comment]
LGTM. Maybe @mfranczy can take a look as well.

**Pull request merged**